plaso/parsers/winreg_plugins/shutdown.py

```diff
@@ -43,7 +44,39 @@ class ShutdownPlugin(interface.WindowsRegistryPlugin):
       interface.WindowsRegistryKeyPathFilter(
           'HKEY_LOCAL_MACHINE\\System\\CurrentControlSet\\Control\\Windows')])
 
-  _UINT64_STRUCT = construct.ULInt64('value')
+  _DEFINITION_FILE = 'filetime.yaml'
+
+  def _ParseFiletime(self, byte_stream):
+    """Parses a FILETIME date and time value from a byte stream.
+
+    Args:
+      byte_stream (bytes): byte stream.
+
+    Returns:
+      dfdatetime.Filetime: FILETIME date and time value or None if no
+          value is set.
+
+    Raises:
+      ParseError: if the FILETIME could not be parsed.
+    """
+    filetime_map = self._GetDataTypeMap('filetime')
+
+    try:
+      filetime = self._ReadStructureFromByteStream(
+          byte_stream, 0, filetime_map)
+    except (ValueError, errors.ParseError) as exception:
+      raise errors.ParseError(
+          'Unable to parse FILETIME value with error: {0!s}'.format(
+              exception))
+
+    if filetime == 0:
+      return None
```

Comment:
Why not return the not-set semantic time here instead of returning None?

Reply:
I considered this but was thinking that I would like to be able to tell when a filetime could not be parsed. For this specific parser is does not matter too much but e.g. in case if more than 1 filetime value would be parsed.